das/metrics.go

```diff
@@ -0,0 +1,132 @@
+package das
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncint64"
+	"go.opentelemetry.io/otel/metric/instrument/syncfloat64"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+var (
+	meter = global.MeterProvider().Meter("das")
+)
+
+type metrics struct {
+	sampled    syncint64.Counter
+	sampleTime syncfloat64.Histogram
+
+	newHead syncint64.Counter
+
+	busyWorkers        asyncint64.Gauge
+	isRunning          asyncint64.Gauge
+	headHeight         asyncint64.Gauge
+	headOfSampledChain asyncint64.Gauge
+}
+
+func (sc *samplingCoordinator) initMetrics() error {
+	sampled, err := meter.SyncInt64().Counter("das_sampled_headers_count",
+		instrument.WithDescription("sampled headers counter"))
+	if err != nil {
+		return err
+	}
+
+	sampleTimeHist, err := meter.SyncFloat64().Histogram("das_sample_time_hist",
+		instrument.WithDescription("header sampling time"))
```

Comment:
I think I would take `hist` out of the name of the meter, and then maybe have the description as "duration of sampling a single header"

Reply:
Maybe it is just me, but when there are 100+ metrics in prometheus with all sorts of types having little `_hist` or `_counter` suffix helps to navigate easier. So it is mostly me hating prometheus query interface. I think we can discuss  how to name metrics so it is consistent across different pkgs.

For this particular one if I saw `das_sample_time`, I would expect a `Gauge`. This could be a personal habbit.